tests/PHPStan/Analyser/data/call-user-func-array.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace CallUserFuncArray;
+
+use function PHPStan\Testing\assertType;
+
+function fun(): int
+{
+	return 3;
+}
+
+function fun3($i, $x, $y): int
+{
+	return 3;
+}
+
+class c {
+	static function m(): string
+	{
+		return 'hello';
+	}
+}
+
+class Foo {
+	function doArray() {
+		assertType('*NEVER*', call_user_func_array('single-arg-only'));
+
+		assertType('int', call_user_func_array('CallUserFuncArray\fun', []));
+
+		assertType('int', call_user_func_array('CallUserFuncArray\fun3', [1 ,2 ,3]));
+
+		assertType('string', call_user_func_array(['CallUserFuncArray\c', 'm'], []));
```

Comment:
Please test this with actual generic functions/methods to see if the arguments are passed correctly I suspect they're not.

Reply:
I had to read this comment several times, until I realized the genius content.

what a smart move to test how parameters are passed to the func call, with a generic parameter... wouldn't have thought about such a smart solution. blows my mind. <3.

added more tests and fixed the extension.